admin/client/Signin/SigninView.js

```diff
@@ -60,7 +60,11 @@ var SigninView = React.createClass({
 				return this.displayError('The email and password you entered are not valid.');
 			} else {
 				// Redirect to where we came from or to the default admin path
-				top.location.href = this.props.from ? this.props.from : Keystone.adminPath;
+				if (Keystone.redirect !== '') {
+					top.location = Keystone.redirect;
```

Comment:
top.location achieves the same thing. I'll change it though to make it more consistent.